### ISSUE TITLE
Slicing to a single value returns number instead of array

### ISSUE THREAD

**Issue by neptunes5thmoon** [constantinpape/z5#102]
When slicing such that the result is a single value, `z5py` returns a number instead of a numpy array. `h5py` and `numpy` keep the number of dimensions.

Here's an example:
```
import numpy as np
import z5py
import h5py

np_arr =np.ones((5,5,5))
np_arr_sl = np_arr[slice(0,1,None),slice(0,1,None),slice(0,1,None)]
print(np_arr_sl.ndim) # 3

hfds = h5py.File('test.h5', 'r')['test_ds'] 
hf_sl = hfds[slice(0,1,None),slice(0,1,None),slice(0,1,None)]
print(hf_sl.ndim) # 3

zfds = z5py.File('test.n5', use_zarr_format=False)['test_ds']
zf_sl = zfds[slice(0,1,None),slice(0,1,None),slice(0,1,None)]
print(zf_sl.ndim) # AttributeError: 'float' object has no attribute 'ndim'
```
And similarly, `h5py` and `numpy` reduce the number of dimensions to 0 for indexing, but still return numpy objects. `z5py` returns the native python type.
```
print(type(hfds[0,0,0])) # <type 'numpy.float64'>
print(type(np_arr[0,0,0])) # <type 'numpy.float64'>
print(type(zfds[0,0,0])) # <type 'float'>
```

Feel free to close if this behavior is intended.

**Comment by clbarnes** [comment]
> z5py returns a number instead of a numpy array

My fault, trying to handle all of numpy's slicing idiosyncrasies is tricky. Should have a fix soon.

**Comment by constantinpape** [comment]
There's a somewhat related issue when indexing with a number, see #103.
@clbarnes If you find the time to look into this, it would be awesome if you could tackle it as well.

**Comment by constantinpape** [comment]
Fixed by #104 thanks to @clbarnes.
